src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+                feature_state = False
+
+            elif feature.state == "conditional":
+                feature_state = {}
+                feature_state["EnabledFor"] = []
+                client_filters = feature.conditions["client_filters"]
+                # client_filters is a list of dictionaries, where all dictionaries have 2 keys - Name and Parameters
+                for filter_ in client_filters:
+                    feature_filter = {}
+                    feature_filter["Name"] = filter_.name
+                    feature_filter["Parameters"] = filter_.parameters
+                    feature_state["EnabledFor"].append(feature_filter)
+
+            feature_entry = {}
+            feature_entry[feature.key] = feature_state
+
+            if format_ == 'json':
+                exported_dict["FeatureManagement"].update(feature_entry)
+
+            elif format_ in ('yaml', 'properties'):
+                featureSetValues["features"].update(feature_entry)
+                exported_dict["feature-management"].update({"featureSet": featureSetValues})
+
+        if format_ == 'properties':
+            exported_dict = __flatten_dict_to_properties(exported_dict)
+
+        return __compact_key_values(exported_dict)
+
+    except Exception as exception:
+        raise CLIError("Failed to export feature flags. " + str(exception))
+
+
+def __convert_features_to_key_value_list(features_dict, format_):
+    # pylint: disable=too-many-nested-blocks
+    key_values = []
+    default_conditions = {'client_filters': []}
+
+    default_value = {
+        "id": "",
+        "description": "",
+        "enabled": False,
+        "conditions": default_conditions
+    }
+    try:
+        if format_ == 'json':
+            for k, v in features_dict.items():
+                key = FEATURE_FLAG_PREFIX + str(k)
+                default_value["id"] = str(k)
+
+                if isinstance(v, dict):
+                    # This is a conditional feature
+                    default_value["enabled"] = True
+                    default_value["conditions"] = {'client_filters': v.get('EnabledFor', [])}
+
+                    if not default_value["conditions"]["client_filters"]:
+                        # We support alternate format for specifying always OFF features
+                        # "FeatureU": {"EnabledFor": []}
+                        default_value["enabled"] = False
+                        default_value["conditions"] = default_conditions
+
+                    else:
+                        # Convert Name and Parameters to lowercase for backend compatibility
+                        for idx, val in enumerate(default_value["conditions"]["client_filters"]):
+                            # each val is a dict with two keys - Name, Parameters
+                            val = {filter_key.lower(): filter_val for filter_key, filter_val in val.items()}
+                            if val["name"] and val["name"].lower() == "alwayson":
```

Comment:
As the schema is fixed, no need to loop val, but explicitly get "Name" and "Parameters". Also we can decide should we fail or log warrning if the schema is not valid(such as name or parameters is missing)